### PR TITLE
Interlok 3226 add 'help' option to interlok boot

### PR DESCRIPTION
**Pull request by jameswickham**
## Motivation

As requested by this Jira ticket: https://adaptris.atlassian.net/browse/INTERLOK-3226

## Modification

CmdLineBootstrap abstract class updated to allow for functionality that will load in a file containing possible command line arguments you can pass it and print it out to the console if "-help" or "--help" arguments are used.

## PR Checklist

- [x] been self-reviewed.
- [x] Added javadocs for most classes and all non-trivial methods
- [n/a] Added supporting annotations (like @XStreamAlias / @ComponentProfile)
- [n/a] Added DefaultValue annotation when there is a default value (e.g. @DefaultValue('true'))
- [n/a] Added validation annotation (@NotNull...) when required and add @Valid when nested class has some validation
- [n/a] Checked that @NotNull and @NotBlank annotations have a meaningful message when appropriate.
- [n/a] Checked that new 3rd party dependencies are appropriately licensed
- [x] Added comments explaining the "why" and the intent of the code wherever it would not be obvious for an unfamiliar reader
- [ ] Added unit tests or modified existing tests to cover new code paths
- [x] Tested new/updated components in the UI and at runtime in an Interlok instance
- [n/a] Reviewed java class members so that missing annotations are added (InputFieldDefault/ComponentProfile etc)
- [x] Checked that javadoc generation doesn't report errors
- [n/a] Checked the display of the component in the UI
- [n/a] Remove any config/license annotations
- [n/a] Check the gradle build file to make sure the dependencies section is more explicit "implementation/api".

## Result

What's the end result for the user

## Testing

How can I test this if I'm reviewing this.

### CODE REVIEW

**Comment by jameswickham** [comment]
@sebastien-belin-adp @aaron-mcgrath-adp @higgyfella 
There are some TBC in the conent on the help file. Thought I should run it past you all to make sure the content of the file is ok and correct before this PR is approved and merged.

Also I can see there is no unit test for this Class at all, wanted to check with you guys before creating one in case there's a reason why.

**Review by sebastien-belin-adp — COMMENTED** [review]
I didn't test it but it looks fine to me except the two comments.
I think there is no tests because for this class because this is what start Interlok and it's probably a bit hard for the benefit of it.
Feel free to have a go at it or maybe just to test your code.

File: interlok-core/src/main/java/com/adaptris/core/management/CmdLineBootstrap.java [hunk 1]
```diff
@@ -146,6 +158,29 @@ protected void logVersionInformation() {
...
   }
+  
+  /**
+   * Print out argument options.
+   * <p>
+   * Note that if the commandline arguments contain "-help or --help" then System.exit() will be invoked after printing out possible cmdline arguments
+   * you can use.
+   * </p>
+   */
+  protected void ArgHelp() throws Exception {
```
[the remark this answers is not in the document]
Comment (jameswickham) [reply interlok-core/src/main/java/com/adaptris/core/management/CmdLineBootstrap.java]: gah what a stupid mistake! corrected now.

**Comment by sebastien-belin-adp** [comment]
@jameswickham I made a commit to remove some javadocs warnings. This hasn't anything to do with what you've done but when I built the jars I got them so that's the opportunity.
Also I've added the -version option to the help file as you forgot it.
I've also added a space before the option name to make it easier to read.

Last, I find it's a bit hard to read on small console window as the help text is long and wrap under the option names (hence why I've added a space).

But I'll approve it anyway as it does the job nicely and we can think about a different format another time or even colors/bold ... ;)

**Review by sebastien-belin-adp — APPROVED** [review]

**Pull request merged**